examples/demoTablesApp/src/App.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Table experiment hole
+ * https://github.com/facebook/react-native
+ *
+ * @format
+ * @flow
+ */
+
+import React, { Fragment, useState, useCallback, useReducer } from 'react'
+import { Button, StatusBar } from 'react-native'
+import { DataTable } from './components/DataTable'
+import { Row } from './components/Row'
+import { Cell } from './components/Cell'
+
+// Configurable constants for demo data volume
+const rowCount = 10
+const columnCount = 4
+
+const baseData = []
+const baseColumns = []
+
+// Make columns definition used in demo app
+for (let index = 0; index < columnCount; index++) {
+  baseColumns.push({ key: `col${index}`, editable: index === columnCount - 1 }) // index === columnCount - 1
+}
+
+// Generate data for use in demo app
+for (let index = 0; index < rowCount; index++) {
+  const rowValues = {}
+  baseColumns.forEach((column, columnIndex) => {
+    rowValues[column.key] = `row${index}Col${columnIndex}`
+  })
+
+  baseData.push({ id: `r${index}`, ...rowValues })
+}
+
+const keyExtractor = item => item.id
+const dataReducer = (data, action) => {
+  switch (action.type) {
+    case 'editRowCell':
+      const { newValue, rowKey, columnKey } = action
+      const rowIndex = data.findIndex(row => keyExtractor(row) === rowKey)
+
+      // Immutable array editing so only the row/cell edited are re-rendered.
+      // If you don't do this, every row will re-render as well as the cell
+      // edited.
+      return data.map((row, index) => {
+        if (index !== rowIndex) {
+          return row
+        }
+        const rowEdited = { ...row }
+        rowEdited[columnKey] = newValue
+        return rowEdited
+      })
+    case 'reverseData':
+      return [...data.reverse()]
+    default:
+      return data
+  }
+}
+
+const App = () => {
+  const [isButtonOof, toggleButton] = useState(false)
+  const [data, dataDispatch] = useReducer(dataReducer, baseData)
+  const columns = baseColumns
+
+  const renderCells = useCallback(
+    (rowData, rowKey) => {
+      return columns.map(col => (
+        <Cell
+          key={col.key}
+          value={rowData[col.key]}
+          rowKey={rowKey}
+          columnKey={col.key}
+          editable={col.editable}
+          dataDispatch={dataDispatch}
+        />
+      ))
+    },
+    [columns, dataDispatch]
+  )
```

Comment:
The current design has `Cell` onChange event call dataDispatch with an action like `{ type: 'editRowCell', newValue, rowKey, columnKey }`. This ends up just being a bit rigid I think...

Ok I experimented with giving a callBack to the cell that calls the dispatch with what ever you want. But then you have another function reference to manage (boom all cells on the row re-render on an edit). useCallback isn't allowed inside a useCallback hook haha. If only there was some kind of "actionCreator" pattern prescribed by some very common library 🤔